feat(dynamodb): Implement the DynamoDB Storage Adapter

thelleboid-tsr opened the pull request:
PR #206 seems stale, so I've rewrritten a new PR from scratch.

thelleboid-tsr committed:
feat(dynamodb): Implement the DynamoDB Storage Adapter

KnorpelSenf commented on packages/dynamodb/README.md:
```diff
@@ -0,0 +1,155 @@
+const bot = new Bot('your-bot-token');
+
+bot.use(
+  session({
```
This leads to a type error, right? So it should either be a JavaScript example or include a setup of the TypeScript types.

thelleboid-tsr replied:
You're right, sorry about that.

I've :
- updated the example to add a valid Typescript code (and checked there were no issues with the typeps)
- added an example for conversation

KnorpelSenf commented on packages/dynamodb/README.md:
```diff
@@ -0,0 +1,155 @@
+
+## TTL (Time To Live)
+
+The adapter automatically sets a TTL of 24 hours for each session. This helps prevent the table from growing indefinitely. You can enable TTL on your DynamoDB table to automatically delete expired items:
```
Does it really? I don't see that in the implementation

thelleboid-tsr replied:
Indeed, it was the original plan but I changed my mind to leave no expiration by default.

I've updated this part (also to explain Grammy handles cleaning data for conversation and may do it for session too).

thelleboid-tsr commented:
Thanks a lot for you quick review!

I've updated the code:
- it's now possible to set a custom primary key (default is set to `sessionKey`)
- it's now possible to set a custom ttl key (default is set to `ttl`)
- I've removed AWS CDK examples in the README.md

Plus fixes in the README.md already commented earlier (I'll resolve Github conversations there, please don't mind if it's not they way to do)

Satont commented:
@thelleboid-tsr thanks for contribution.

I'll merge this pr today and release version.

/cc @KnorpelSenf

Satont committed:
Merge branch 'main' into feat/dynamodb-storage

Satont commented:
Thats awesome. I'll fix pnpm lockfile conflict in main branch.
